### PR TITLE
refactor(location-redirection-utils: Deprecate package and move API elements to `container-loader`

### PR DESCRIPTION
**Pull request by Josmithr**
This package is not used internally, and is only used by a single partner (currently). The package exports 2 items, `isLocationRedirectionError` and `resolveWithLocationRedirectionHandling`. These have been moved to `container-loader`, and the corresponding exports in `location-redirection-utils` have been marked as deprecated.

[AB#5070](https://dev.azure.com/fluidframework/internal/_workitems/edit/5070)

### CODE REVIEW

**Comment by jatgarg** [comment]
I think the right place would be inside container-loader package. Maybe create a separate folder inside that package.
container-loader/src/location-redirection-utils/....

File: packages/loader/container-loader/src/location-redirection-utilities/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export {
```
Comment (jatgarg) [inline packages/loader/container-loader/src/location-redirection-utilities/index.ts:6]: There is no need of 2 index.ts files. Just export this from the outside main one.

Comment (Josmithr) [reply packages/loader/container-loader/src/location-redirection-utilities/index.ts:6]: Drilling into directories is generally considered bad practice, and our linter rules explicitly forbid it outside of particular contexts.

File: packages/loader/location-redirection-utils/package.json [hunk 2]
```diff
@@ -30,10 +30,7 @@
 		"lint:fix": "npm run prettier:fix && npm run eslint:fix",
 		"prettier": "prettier --check . --ignore-path ../../../.prettierignore",
 		"prettier:fix": "prettier --write . --ignore-path ../../../.prettierignore",
-		"test": "npm run test:mocha",
-		"test:coverage": "c8 npm test",
-		"test:mocha": "mocha --ignore 'dist/test/types/*' --recursive dist/test -r node_modules/@fluidframework/mocha-test-setup",
-		"test:mocha:verbose": "cross-env FLUID_TEST_VERBOSE=1 npm run test:mocha",
+		"test": "echo \"Tests have been removed. This package is scheduled for deletion.\" && exit 1",,
```
Comment (Josmithr) [inline packages/loader/location-redirection-utils/package.json:33]:
```suggestion
		"test": "echo \"Tests have been removed. This package is scheduled for deletion.\" && exit 1",
```

**Review by jatgarg — APPROVED** [review]

**Pull request merged**